app/services/RegisteredBusinessService.scala

```diff
@@ -88,7 +89,8 @@ trait RegisteredBusinessService {
       }
     }
 
-    val agentArn = user.principal.accounts.agent.flatMap(_.agentBusinessUtr.map(_.value))
+    val agentArn = user.enrolments.getEnrolment("HMRC-AGENT-AGENT")
```

Comment:
You made a getArn function that could replace this (and other similar instances of this)